src/client.ts

```diff
@@ -5000,7 +5000,8 @@ export class MatrixClient extends TypedEventEmitter<EmittedEvents, ClientEventHa
         });
 
         if (!unthreaded) {
-            const isThread = !!event.threadRootId;
+            // A thread cannot be just a thread root and a thread root can only be read in the main timeline
+            const isThread = !!event.threadRootId && !event.isThreadRoot;
```

Comment:
I think it might be nice if we had a `event.isThreadReply` boolean that handled this complexity in a single place

Reply:
We do, `isRelation(RelationType.Thread)` but that doesn't include 2nd order relations which we do want to include here. Reactions to thread responses aren't "thread replies" yet still want RRs

Reply:
Whatever we want to call this logic `!!event.threadRootId && !event.isThreadRoot;` feels like it could be nicely encapsulated with a description of the intracies.

"thread replies" still feels applicable and a good enough name for reactions in this scenario but if there is a better name, that would be nice. I don't know how reactions to a thread root exactly works though. And those kind of details could be nicely explained there.